lab_solutions/lab03_3_rational_number_class.py

```diff
@@ -96,24 +93,28 @@ def q(self, value):
         self._q = value
         if value < 0:
             self.switch_signs()
+        self.normalize_fraction()
 
-    @staticmethod
-    def validate_numerator(p):
+    @classmethod
+    def validate_numerator(cls, p):
         if not isinstance(p, int):
             raise RationalNumberValueError('Numerator has to be an Integer')
 
-    @staticmethod
-    def validate_denominator(q):
+    @classmethod
+    def validate_denominator(cls, q):
         if not isinstance(q, int):
             raise RationalNumberValueError('Denominator has to be an Integer')
         if q == 0:
             raise RationalNumberValueError('Cannot set denominator to 0')
 
-    @staticmethod
-    def from_str(input_str):
+    @classmethod
+    def from_str(cls, input_str):
         p, q = input_str.split('/')
         return RationalNumber(int(p), int(q))
```

Comment:
The point of a classmethod is to use the class object. This would work for `RationalNumber` subclasses not just `RationalNumber`:
```suggestion
        return cls(int(p), int(q))
```